### DISCUSSION TITLE
How to migrate data to new instance

### DISCUSSION THREAD

**Discussion by autoantwort** [dani-garcia/vaultwarden#4333] [Q&A] [answered]
I currently have a running instance create with https://tteck.github.io/Proxmox/ which has version 2.25.1. 
I now wanted to migrate to the docker version of vaultwarden. So I have run `docker run -d --name vaultwarden -v /vw-data/:/data/ -p 80:80 -e 'ADMIN_TOKEN=<old_admin_token>' vaultwarden/server:latest` and copied the old data folder to `/vw-data`. But it seems that the container deletes the existing files and creates new empty files. 
Does anybody know how to migrate the data from an existing instance to a new instance that utilizes docker?

**Comment by stefan0xC** [comment]
> So I have run `docker run -d --name vaultwarden -v /vw-data/:/data/ -p 80:80 -e 'ADMIN_TOKEN=<old_admin_token>' vaultwarden/server:latest` and copied the old data folder to `/vw-data`.

Did you do it in that order?

**Reply by autoantwort** [reply, marked as answer]
Ok I don't know what I did wrong, but the following works: 
1. `docker stop vaultwarden`
2. `rm -r /vw-data`
3. `cp -r /old-data /vw-data`
4. `docker start vaultwarden`

**Comment by alex3025** [comment]
There's a way to migrate everything BUT the Vaultwarden settings (the ones you edit in `/admin`)? 

I'm the only user so maybe I could just use the normal _Export (as JSON)_ feature? Is that enough?

**Reply by stefan0xC** [reply]
If you copy the files and the data in the database you can easily migrate everything the same way you would when you'd restore your data from a backup (cf. [backing up your vault](https://github.com/dani-garcia/vaultwarden/wiki/Backing-up-your-vault) for more information).

And yes, [exporting the data](https://bitwarden.com/help/export-your-data/) from an old account and then [importing it](https://bitwarden.com/help/import-data/) into a new account would also be an option. Just be aware that not all data can be migrated that way (at least last I checked it was not possible to import file attachments or password history, etc).

**Reply by alex3025** [reply]
> If you copy the files and the data in the database you can easily migrate everything the same way you would when you'd restore your data from a backup (cf. [backing up your vault](https://github.com/dani-garcia/vaultwarden/wiki/Backing-up-your-vault) for more information).

But this will copy also the admin settings, right?

**Reply by stefan0xC** [reply]
Yep, but you can leave out the `data/config.json` if you don't want to copy it. (I have not configured my instance via the `/admin` panel only via environment variables, so I don't have that file.)

**Reply by alex3025** [reply]
So I could copy everything but the `config.json` file to "reset" the admin settings?

**Reply by stefan0xC** [reply]
If you only want to reset the settings you changed via the `/admin` panel, you can also delete (or move) the `config.json`. I'd recommend reading the [configuration overview](https://github.com/dani-garcia/vaultwarden/wiki/Configuration-overview) if you are confused about how Vaultwarden is configured.
